OrbitClientModel/CaptureSerializer.cpp

```diff
@@ -62,6 +62,17 @@ CaptureInfo GenerateCaptureInfo(
   capture_info.mutable_thread_names()->insert(capture_data.thread_names().begin(),
                                               capture_data.thread_names().end());
 
+  for (const auto& tid_and_thread_state_slices : capture_data.thread_state_slices()) {
+    // Note that thread state slices are saved in their original order only among the same thread,
+    // but all slices related to the same thread are saved sequentially. This might not be desired
+    // if the capture is opened in a streaming fashion.
```

Comment:
Well we store them by thread in our data model. So I think it would be the easiest way to simulate something like real-time, if we just seriealize them also by thread, and then have multiple threads reading the serialized file.

Of curse, we could also just serialize all events together sorted by timestamp.

Reply:
I think it's overkill to plan for reading with multiple threads.

The reasonable thing to do here is to sort by timestamp, but I didn't do it because it adds an overhead that doesn't provide any value until *all* events are sorted by timestamp. Until then, having states come in order when for example they all come before callstacks which all come before (or after?) timers is not useful.

Reply:
agree. lets postpone this.